### PR TITLE
Use libepoxy

### PR DESCRIPTION
**Pull request by yshui**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,14 @@
...
 * Allow `corner-radius-rules` to override `corner-radius = 0`. Previously setting corner radius to 0 globally disables rounded corners. (#1170)
 
+## Build changes
+
+* `picom` now depends on `libepoxy` for OpenGL symbol management.
+
 ## Bug fixes
 
 * Workaround a NVIDIA problem that causes high CPU usage after suspend/resume (#1172, #1168)
+* Fix building on OpenBSD (#1189 #1188)
```
Comment (absolutelynothelix) [inline CHANGELOG.md]:
```suggestion
* Fix building on OpenBSD (#1189, #1188)
```

File: flake.nix [hunk 2]
```diff
@@ -24,16 +24,20 @@
...
-        pkgs.llvmPackages_17.clang-unwrapped.python
-      ];
+      buildInputs = defaultPackage.buildInputs ++ (with pkgs; [
+        clang-tools_17
+        llvmPackages_17.clang-unwrapped.python
+      ]);
       hardeningDisable = [ "fortify" ];
+      shellHook = ''
+        export LD_LIBRARY_PATH+=":/run/opengl-driver/lib"
+      '';
```
Comment (absolutelynothelix) [inline flake.nix]: afaiu it's a workaround and i think i saw it noted in one of the commits' messages. maybe it's worth a mention here as well if comments are allowed in flakes?

File: src/backend/gl/egl.c [hunk 3]
```diff
@@ -422,66 +395,22 @@ struct backend_operations egl_ops = {
...
 
 void eglext_init(EGLDisplay dpy) {
 	if (eglext.initialized) {
 		return;
 	}
 	eglext.initialized = true;
-#define check_ext(name) eglext.has_##name = egl_has_extension(dpy, #name)
+#define check_ext(name)                                                                  \
+	eglext.has_##name = epoxy_has_egl_extension(dpy, #name);                         \
+	log_info("Extension %s - %s", #name, eglext.has_##name ? "present" : "absent");
```
Comment (absolutelynothelix) [inline src/backend/gl/egl.c]: honestly, i don't like the format of this message but i can't think of anything really better right now.

File: src/backend/gl/glx.c [hunk 4]
```diff
@@ -545,62 +545,17 @@ struct backend_operations glx_ops = {
...
 
 void glxext_init(Display *dpy, int screen) {
 	if (glxext.initialized) {
 		return;
 	}
 	glxext.initialized = true;
-#define check_ext(name) glxext.has_##name = glx_has_extension(dpy, screen, #name)
+#define check_ext(name)                                                                  \
+	glxext.has_##name = epoxy_has_glx_extension(dpy, screen, #name);                 \
+	log_info("Extension " #name " - %s", glxext.has_##name ? "present" : "absent")
```
Comment (absolutelynothelix) [inline src/backend/gl/glx.c:557]: same as above.

File: src/meson.build [hunk 5]
```diff
@@ -23,7 +23,7 @@ required_xcb_packages = [
 # Some XCB packages are here because their versioning differs (see check below).
 required_packages = [
 	'pixman-1', 'x11', 'x11-xcb', 'xcb-image', 'xcb-renderutil', 'xcb-util',
-	'xext'
+	'xext',
```
Comment (absolutelynothelix) [inline src/meson.build]: huh?

File: src/meson.build [hunk 6]
```diff
@@ -59,7 +59,10 @@ endif
 
 if get_option('opengl')
 	cflags += ['-DCONFIG_OPENGL', '-DGL_GLEXT_PROTOTYPES']
```
Comment (absolutelynothelix) [inline src/meson.build]: do we still need `-DGL_GLEXT_PROTOTYPES`?

File: src/meson.build [hunk 7]
```diff
@@ -59,7 +59,10 @@ endif
...
-	deps += [dependency('gl', required: true), dependency('egl', required: true), dependency('threads', required:true)]
```
Comment (absolutelynothelix) [inline src/meson.build]: do we still explicitly depend on the `libgl` and `libegl`?

**Review by absolutelynothelix — COMMENTED** [review]
looks good, builds and works for me. a few things may need to be addressed before merging.

File: src/meson.build (hunk 5 above)
Comment (yshui) [reply src/meson.build]: trailing commas are good

File: src/meson.build (hunk 6 above)
Comment (yshui) [reply src/meson.build]: good catch

**Comment by yukiteruamano** [comment]
@yshui @absolutelynothelix I'm going to try the changes in OpenBSD, there shouldn't be any problems with using libepoxy, but it's better to clear up any doubts

File: src/backend/gl/glx.c (hunk 4 above)
Comment (absolutelynothelix) [reply src/backend/gl/glx.c:557]: also, this line is a bit different compared to the one above. let's make them the same at least (don't miss the semicolon at the end!).

Comment (yshui) [reply src/backend/gl/glx.c:557]: no i need to remove the semicolon on the other line.

**Comment by absolutelynothelix** [comment]
@yukiteruamano, we now have an openbsd ci so picom should at least always build there. sorry for the inconvenience!

**Comment by yukiteruamano** [comment]
Yeah, I see that and I test the last changes, compile without issues

**Pull request merged**